exercises/concept/lucians-luscious-lasagna/.docs/introduction.md

```diff
@@ -1,104 +1,51 @@
-
-```rust
-// Automatically inferred type
-let fingers = 10;
-```
-
-Functions are _items_. Where bindings typically refer to a particular value, items refer to a unit of code organization, typically a function or a module, which is available throughout the lifetime of the program. A function automatically returns the result of its last expression. A function may have 0 or more parameters, which are bindings with a lifetime of the function call.
-
-Type inference is theoretically possible for functions, but is disabled as an intentional language design choice. While this means that you need to spend a little more time when writing code to specify precisely what a function's input and output types are, you save the time when you're reading the code, because all the input and output types are explicitly defined.
+Functions are a fundamental building block of Rust programs.
+Let's look at a function to get an idea of what's going on:
 
 ```rust
 fn add(x: i32, y: i32) -> i32 {
   x + y
 }
 ```
 
-Invoking a function is done by specifying its name followed by parentheses. If the function requires parameters, an argument must be specified for each within the parentheses.
-
-```rust
-let five = add(2, 3);
-```
-
-If a binding's type cannot be inferred, the compiler will report an error. To fix this, add an explicit type annotation to the binding.
-
-```rust
-// Explicit type annotation
-let fingers: i32 = 10;
-```
-
-Items in Rust can be used before or after they are defined, because they have a static lifetime. Bindings, on the other hand, can only be used _after_ they have been defined. Using a binding before it has been defined results in a compile error.
-
-```rust
-fn main() {
-    // `fn add` hasn't yet been defined, but that's perfectly ok
-    dbg!(add(3, 4));
-}
-
-fn add(x: i32, y: i32) -> i32 {
-  x + y
-}
-```
+The first keyword is `pub`, a visibility modifier.
+It is optional and you will learn about it later, no need to worry about it now.
 
-```rust
-// this won't compile; `a` is used before its binding is defined
-let b = a;
-let a = x + y;
-```
+The keyword `fn` and the name of the function are mandatory.
 
-Rust uses curly braces (`{}`) to define a scope. A binding defined within a scope can't escape from it.
+A function may have zero or more parameters enclosed in parentheses `()`.
+A parameter consist of its name and type, separated by a colon.
+Multiple parameters are separated by commas.
 
-```rust
-let a = 1;
-dbg!(a); // 1
-{
-    // Here, we re-bind `a` to a new value, which is still immutable.
-    // This technique is called _shadowing_. The new binding is constrained to
-    // this anonymous scope. Outside this scope, the previous binding still
-    // applies.
-    let a = 2;
-    let b = 3;
-    dbg!(a, b); // 2, 3
-}
-// can't use `b` anymore because it is out of scope
-// dbg!(b);
+The return type is specified after an arrow `->`.
+It is optional if the function doesn't return anything.
```

Comment:
One might wonder: if the function does not return anything, how could one give its (presumably nonexistent) return type?

Reply:
Good catch. I decided to change the wording so it seems like that's not possible. No need to confuse students with the empty tuple at this point.